onnxruntime/core/providers/shared/utils/utils.cc

```diff
@@ -166,6 +166,13 @@ std::vector<float> NodeAttrHelper::Get(const std::string& key, const std::vector
   return std::vector<float>{source.cbegin(), source.cend()};
 }
 
+std::optional<int32_t> NodeAttrHelper::Get(const std::string& key) const {
+  if (!HasAttr(key))
+    return std::nullopt;
+
+  return SafeInt<int32_t>(node_attributes_.at(key).f());
```

Comment:
`node_attributes_.at(key).f()` returns a float attribute, right? `i()` returns an integer attribute.

for Split, `num_outputs` is specified to be an int.

we should consider how this new interface can be extended to support other attribute types. unfortunately we can't have additional function signatures which only differ in return type.

one idea:
```c++
template <typename T>
std::optional<T> Get(const std::string& key) const;
```

or a simpler option could be:
```c++
std::optional<int64_t> GetInt(const std::string& key) const {
  if (!HasAttr(key))
    return std::nullopt;
  return node_attributes_.at(key).i();
}
```

Reply:
oops, changed to the second option here.